.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,174 @@
+          secret = ")9gk_4A6KrXz9D2u'0@MPea*sd6W'%@5MAWpWWJ3P3EqW~qB!!(Vd$FhNc*eU4vG"
+          time_to_live = "8760h"
+
+          [apiserver]
+            bind = "0.0.0.0"
+            port = 9997
+            use_tls = false
+
+          [database]
+            backend = "sqlite3"
+            passphrase = "shreotsinWadquidAitNefayctowUrph"
+            [database.sqlite3]
+              db_file = "/etc/garm/garm.db"
+
+          [[provider]]
+            name = "lxd_local"
+            provider_type = "lxd"
+            description = "Local LXD installation"
+            [provider.lxd]
+              unix_socket_path = "/var/snap/lxd/common/lxd/unix.socket"
+              include_default_profile = false
+              instance_type = "container"
+              secure_boot = false
+              project_name = "default"
+              url = ""
+              client_certificate = ""
+              client_key = ""
+              tls_server_certificate = ""
+              [provider.lxd.image_remotes]
+                [provider.lxd.image_remotes.ubuntu]
+                  addr = "https://cloud-images.ubuntu.com/releases"
+                  public = true
+                  protocol = "simplestreams"
+                  skip_verify = false
+                [provider.lxd.image_remotes.ubuntu_daily]
+                  addr = "https://cloud-images.ubuntu.com/daily"
+                  public = true
+                  protocol = "simplestreams"
+                  skip_verify = false
+                [provider.lxd.image_remotes.images]
+                  addr = "https://images.linuxcontainers.org"
+                  public = true
+                  protocol = "simplestreams"
+                  skip_verify = false
+          EOF
+          sudo mv config.toml /etc/garm/config.toml
+          sudo chown -R garm:garm /etc/garm /var/log/garm
+
+          sudo mv ./bin/* /usr/local/bin/
+          sudo cp ./contrib/garm.service /etc/systemd/system/garm.service
+
+          sudo systemctl daemon-reload
+          sudo systemctl start garm
+
+          wait_open_port() {
+              local ADDRESS="$1"
+              local PORT="$2"
+              local TIMEOUT=30
+              SECONDS=0
+              while true; do
+                  if [[ $SECONDS -gt $TIMEOUT ]]; then
+                      echo "ERROR: Port $PORT didn't open at $ADDRESS within $TIMEOUT seconds"
+                      return 1
+                  fi
+                  nc -v -w 5 -z "$ADDRESS" "$PORT" &>/dev/null && break || sleep 1
+              done
+              echo "Port $PORT at address $ADDRESS is open"
+          }
+
+          wait_open_port 127.0.0.1 9997
+
+      - name: Test GARM API with garm-cli tool
+        shell: bash
+        run: |
+          set -e
+          set -x
+          set -o pipefail
+
+          GARM_URL="http://127.0.0.1:9997"
+          ADMIN_USERNAME='admin'
+          ADMIN_PASWORD='Ma*jirksiCr9esrT8DGQPrfx*XKnBVZo'
+
+          garm-cli init --debug \
```

Comment:
While it's great that the CLI gets some tests as well, I think that it would be better if you used the code you already have here:

https://github.com/mihaelabalutoiu/garm-test-client/blob/main/main.go

We can set up a nice integration test suite, that spins up the GARM API and daemon loops, then run the code at the above link to test everything.

This way we can easily see if the output from the API matches expectations. We also end up testing both GARM and the client library.

The CLI is just a wrapper on top of the client library anyway (with some pretty tables).

Reply:
we can also just set up garm as a service like you did above and run the code at the link against it. Both options are fine.

Reply:
Sounds good! I will setup `garm` as a service and run that code.  Thank you :)